File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        start=0
        res=0
        answer=0
        for i in range(len(s)):
            res+=abs(ord(s[i])-ord(t[i]))
            while res > maxCost:
                res-=abs(ord(s[start])-ord(t[start]))
                start+=1
            answer=max(answer,i-start+1)
        return answer

File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int start=0;
        int answer=0;
        int res=0;
        for(int i=0;i<s.size();i++){
            res+=abs(s[i]-t[i]);
            while(res > maxCost){
                res-=abs(s[start]-t[start]);
                start+=1;
            }
            answer=max(answer,i-start+1);
        }
        return answer;
    }
};